status/api/provider.nim

```diff
@@ -23,3 +28,59 @@ proc callRpc*(self: StatusObject, rpcMethod: string, params: JsonNode):
     return CallRpcResult.ok await self.web3.callRpc(rpcMethod, params)
   except CatchableError as e:
     return CallRpcResult.err e.msg
+
+proc signTransaction(tr: var Transaction, pk: PrivateKey) =
+  let h = tr.txHashNoSignature
+  let s = sign(pk, SkMessage(h.data))
+
+  var r = toRaw(s)
+  let v = r[64]
+
+  tr.R = fromBytesBE(UInt256, r.toOpenArray(0, 31))
+  tr.S = fromBytesBE(UInt256, r.toOpenArray(32, 63))
+
+  tr.V = int64(v)
+
+  if tr.txType == TxType.TxLegacy:
+    tr.V += 27 # TODO! Complete this
+
+proc sendTransaction*(self: StatusObject, fromAddress: EthAddress, transaction: Transaction, password: string, dir: string):
+  Future[SendTransactionResult] {.async.} =
+  if not self.isLoggedIn:
+    return SendTransactionResult.err "Not logged in. Must be logged in"
+
+  try:
+    let accounts = self.userDb.getWalletAccounts()
+    let account = accounts.filter(x => x.address == Address(fromAddress))
+    if account.len == 0:
+      return SendTransactionResult.err "Wallet address not found"
```

Comment:
It was discovered today that when using `getWalletAccount` in `status/private/accounts/accounts.nim` (as this PR does presently), then because of the [`AND wallet = 0`](https://github.com/status-im/nim-status/blob/master/status/private/accounts/accounts.nim#L112) condition, it's not possible get the account that corresponds to the `Status account`.

Here's how to replicate:
* delete the client's data directory, if it exists
* build and run the client
* do `/create 1` and `/login 1 1`
* do `/wallets` and copy/paste address for `Status account` into:
```
/trx [address] 0x0000000000000000000000000000000000000000 0 1 255 10000 0x7f7465737432000000000000000000000000000000000000000000000000000000600057 0 1
```

Then run that command. In the TUI an error will be displayed: `Error Wallet address not found`.

Next do `/addwallet 1`, get the address for the new wallet from `/wallets` and run the `/trx ...` command with that address. An error will be printed; you'll see something like:
```
Error {"code":-32602,"message":"transaction could not be decoded: chainId is required for EIP-2718 style signatures"}
```

Which means the command and nim-status lib did work as expected.

Maybe `getWalletAccount` should be checking for `AND chat = 0`?

Reply:
status-go checks `chat = 0` in its `GetWalletAddresses` function:

https://github.com/status-im/status-go/blob/develop/multiaccounts/accounts/database.go#L634-L635

So we can/should do likewise. I'm going to merge this PR and create another re: the fix for `getWalletAccount`.

Reply:
See #288.